velox/common/base/SpillConfig.h

```diff
@@ -113,13 +116,17 @@ struct SpillConfig {
   /// memory usage.
   int32_t spillableReservationGrowthPct;
 
-  /// Used to calculate spill partition number.
+  /// The start partition bit offset of the top (the first level) partitions.
   uint8_t startPartitionBit;
 
   /// Used to calculate the spill hash partition number for hash join with
   /// 'startPartitionBit'.
   uint8_t joinPartitionBits;
 
+  /// Used to calculate the spill partition number of the hash table in
+  /// RowNumber with 'startPartitionBit'.
+  uint8_t rowNumberPartitionBits;
```

Comment:
Curious, why do we want separate variable for join and rowNumber?

Reply:
This would enable us to fine-tune the number of spill partitions per operator.

Reply:
Why would we want to do that?

Reply:
At first, I want to differentiate the `joinPartitionBits` and `rowNumberPartitionBits` when calling the function `exceedSpillLevelLimit`, so I added the config `rowNumberPartitionBits`.
e.g.
- HashBuild
  - spillConfig.exceedSpillLevelLimit(startBit, **spillConfig.joinPartitionBits**)
- RowNumber
  - spillConfig.exceedSpillLevelLimit(startBit, **spillConfig.rowNumberPartitionBits**)

Reply:
> Yes, I don't see any reason for setting different values of numPartitionBits for different operators and I'm a bit concerned about having inconsistent values.

Got it, I will unify them as `numPartitionBits`.